enhancements/machine-api/control-plane-machine-set.md

```diff
@@ -0,0 +1,1090 @@
+In particular it will also allow them to have different specs across their control plane.
+While in the normal case this is discouraged, this could be used for short periods to test new configuration before
+rolling the updated configuration to all Control Plane Machines.
+
+Note that the OnDelete strategy is similar to the RollingUpdate strategy in that it needs to be able to add new
+Machines to the cluster to function. It can be thought of as a slower, more controlled RollingUpdate strategy where the
+user decides when each Control Plane Machine is replaced, and marks it for replacement by deleting it.
+Otherwise the replacement is identical to the RollingUpdate strategy.
+
+#### Protecting etcd quorum during update operations
+
+With the introduction of the [etcd protection enhancement](https://github.com/openshift/enhancements/pull/943), the
+ControlPlaneMachineSet does not need to observe anything etcd related during scaling operations.
+In particular, it is the etcd operators responsibility to add Machine deletion hooks to prevent Control Plane Machines
+from being removed until they are no longer needed.
+When the `ControlPlaneMachineSet` controller observes that a newly created Machine is ready, it will delete the old
+Control Plane Machine signalling to the etcd operator to switch the membership of the etcd cluster between the old and
+new instances.
+Once the membership has been switched, the Machine deletion hook will be removed on the old Machine, allowing it to be
+removed by the Machine controller in the normal way.
+
+#### Removing/disabling the ControlPlaneMachineSet
+
+As some users may want to remove or disable the ControlPlaneMachineSet, a finalizer will be placed on the
+`ControlPlaneMachineSet` to allow the controller to ensure a safe removal of the ControlPlaneMachineSet, while leaving
+the Control Plane Machines in place.
+
+Notably it will need to ensure that there are no owner references on the Machines pointing to the ControlPlaneMachineSet
+instances. This will prevent the garbage collector from removing the Control Plane Machines when the
+`ControlPlaneMachineSet` is deleted.
+
+If users later wish to reinstall the `ControlPlaneMachineSet`, they are free to do so.
+
+Note: Owner references will be added to the Control Plane Machines to identify to other components that a controller is
+managing the state of these Machines. This allows other systems such as the MachineHealthCheck to identify that if they
+were to make an action on the Machine, the Control Plane Machine Set Operator will react to that action.
+
+#### Installing a ControlPlaneMachineSet within an existing cluster
+
+When adding a ControlPlaneMachineSet to a existing cluster, the end user will need to define the
+`ControlPlaneMachineSet` resource by copying the existing Control Plane Machine ProviderSpecs.
+Once this is copied, they should remove the failure domain and add the desired failure domains to the `FailureDomains`
```

Comment:
> remove the failure domain and add the desired failure domains to the `FailureDomains`

Just a bit confused by this - where are they removing the failure domain?

Reply:
Yeah that's it, to give an explicit example, and I'm not sure how to word this well but:
- User copies providerSpec from an existing ControlPlaneMachine
- They paste that onto the new `ControlPlaneMachineSet.Spec.Template.machines_v1beta1_machine_openshift_io.Spec.ProviderSpec.Value`
- In the case of AWS, they now delete `Placement.AvailabilityZone` and `Subnet` from the `ProviderSpec.Value`
- They now add their failure domains to `ControlPlaneMachineSet.Spec.Template.machines_v1beta1_machine_opneshift_io.FailureDomains.AWS`
- The failure domains will look like:
```
failureDomains:
  platform: aws
  aws:
  - placement:
      availabilityZone: us-east-1a
    subnet:
      type: filters
      filters:
        name: "tag:Name"
        values:
        - subnet-us-east-1a
  - placement:
      availabilityZone: us-east-1b
    subnet:
      type: filters
      filters:
        name: "tag:Name"
        values:
        - subnet-us-east-1b
  - placement:
      availabilityZone: us-east-1c
    subnet:
      type: filters
      filters:
        name: "tag:Name"
        values:
        - subnet-us-east-1c
```

Reply:
Perfect, that's very helpful!